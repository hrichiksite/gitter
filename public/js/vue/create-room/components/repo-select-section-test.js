const mount = require('../../__test__/vuex-mount');
const { default: RepoSelectSection } = require('./repo-select-section.vue');

describe('RepoSelectSection', () => {
  it('matches snapshot', () => {
    const { wrapper } = mount(RepoSelectSection);
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('GitHub user', () => {
    it('no private repo scope', () => {
      const { wrapper } = mount(RepoSelectSection, {}, store => {
        store.state.user = {
          providers: ['github']
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('with private repo scope granted', () => {
      const { wrapper } = mount(RepoSelectSection, {}, store => {
        store.state.user = {
          providers: ['github'],
          scopes: {
            private_repo: true
          }
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
