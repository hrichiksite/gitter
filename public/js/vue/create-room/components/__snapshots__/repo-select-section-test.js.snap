// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoSelectSection GitHub user no private repo scope 1`] = `
<section>
  <div
    class="repo-select-section"
  >
    <h3
      class="section-heading"
    >
      <label>
        Repo association (optional)
      </label>
    </h3>
     
    <select-dropdown-stub
      filterplaceholder="Filter repos"
      filtertext=""
      items=""
    />
     
    <p
      class="repo-select-information"
    >
      
        If you don't see your repo listed, try
        
      <a
        href="/login/upgrade?scopes=repo"
      >
        allowing private repo access
      </a>
      . Also see our
        documentation section on
        
      <a
        href="https://gitlab.com/gitlab-org/gitter/webapp/-/blob/develop/docs/faq.md#why-isnt-my-github-organisation-or-repos-appearing"
        target="_blank"
      >
        "Why isn't my GitHub organisation or repos appearing?"
      </a>
      .
      
    </p>
  </div>
</section>
`;

exports[`RepoSelectSection GitHub user with private repo scope granted 1`] = `
<section>
  <div
    class="repo-select-section"
  >
    <h3
      class="section-heading"
    >
      <label>
        Repo association (optional)
      </label>
    </h3>
     
    <select-dropdown-stub
      filterplaceholder="Filter repos"
      filtertext=""
      items=""
    />
     
    <p
      class="repo-select-information"
    >
      
        Private repo access is granted. If you don't see your repo listed, see our documentation
        section on
        
      <a
        href="https://gitlab.com/gitlab-org/gitter/webapp/-/blob/develop/docs/faq.md#why-isnt-my-github-organisation-or-repos-appearing"
        target="_blank"
      >
        "Why isn't my GitHub organisation or repos appearing?"
      </a>
      .
      
    </p>
  </div>
</section>
`;

exports[`RepoSelectSection matches snapshot 1`] = `
<section>
  <div
    class="repo-select-section"
  >
    <h3
      class="section-heading"
    >
      <label>
        Repo association (optional)
      </label>
    </h3>
     
    <select-dropdown-stub
      filterplaceholder="Filter repos"
      filtertext=""
      items=""
    />
     
    <!---->
  </div>
</section>
`;
